lib/kits/core-ui/paged-publication.js

```diff
@@ -0,0 +1,206 @@
+        ),
+        shoppingList: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-shopping-list-template'
+        ),
+        shoppingListCounter: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-shopping-list-counter-template'
+        ),
+        pageList: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-page-list-template'
+        ),
+        menuContainer: document.getElementById(
+            'sgn-sdk-paged-publication-viewer-menu-container-template'
+        )
+    };
+
+    const renderMainContainer = () => {
+        const mainContainerEl = document.querySelector(scriptEls.mainContainer);
+
+        PagedPublicationComponents.MainContainer({
+            template: customTemplates.mainContainer,
+            shoppingListCounterTemplate: customTemplates.shoppingListCounter,
+            el: mainContainerEl,
+            scriptEls
+        }).render();
+    };
+    renderMainContainer();
+
+    const pagedPublicationObj = {
+        render: async () => {
+            if (Object.keys(options).length === 0) {
+                await pagedPublicationObj.setOptions();
+            }
+
+            pagedPublicationObj.start();
+
+            pagedPublicationObj.renderShoppingList();
+            pagedPublicationObj.renderMenuPopup();
+        },
+
+        renderShoppingList: () => {
+            PagedPublicationComponents.ShoppingList({
+                template: customTemplates.shoppingList
+            }).render();
+        },
+
+        renderMenuPopup: () => {
+            const menuBtn = document.querySelector('.sgn__nav-menu-btn');
+            menuBtn.addEventListener('click', (e) => {
+                e.preventDefault();
+                PagedPublicationComponents.MenuPopup({
+                    configs: options,
+                    sgnData,
+                    sgnViewer,
+                    templates: customTemplates
+                }).render();
+            });
+        },
+
+        animateShoppingListCounter: () => {
+            const shoppingListCounter = document.querySelector(
+                '.sgn__offer-shopping-list-counter'
+            );
+
+            shoppingListCounter?.classList.remove('sgn-animate-bounce');
+            shoppingListCounter?.classList.add('sgn-animate-bounce');
+        },
+
+        setOptions: async (opts = {}) => {
+            options = {
+                el: document.querySelector('.sgn__pp'),
+                eventTracker: SGN.config.get('eventTracker'),
+                pageId:
+                    opts.pageId ||
+                    `page${SGN.util.getQueryParam(scriptEls.pageIdParam) || 1}`,
+                id:
+                    opts.id ||
+                    SGN.util.getQueryParam(scriptEls.publicationIdParam) ||
+                    scriptEls.publicationId ||
+                    '',
+                businessId: opts.businessId || scriptEls.businessId
```

Comment:
You could probably extract all these properties as free variables in the `pagedPublication` closure